keps/sig-network/3015-node-level-topology/README.md

```diff
@@ -0,0 +1,502 @@
+correctly based on this feature. The user would need to packet-sniff,
+or check individual endpoint logs, or look at the iptables rules
+generated by kube-proxy.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Services using node-level topology should be as reliable as services
+_not_ using node-level topology, and should have lower latency.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [X] Other
+
+There are no existing metrics for determining the availability or
+response time of individual Services, and this KEP does not add any.
+In general, the operator cannot know what response times the user
+expects from their Services anyway.
+
+The developer may be able to generate their own metrics based on logs
+from the endpoint pods and/or logs generated by the service's clients;
+but this involves user-owned code, not kubernetes code, on both ends.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Not really. We could expose the number of node-local-topology services
+on each node, but it's not clear that that would be useful.
+
+### Dependencies
+
+###### Does this feature depend on any specific services running in the cluster?
+
+It depends on a service proxy which understands the feature. We will
+update `kube-proxy` ourselves, but network plugins / kubernetes
+distributions that ship their own alternative service proxies will
+also need to be updated to support the feature before their users can
+benefit from it.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+It might add a new field to EndpointSliceHints (which would be unset
+in most EndpointSlices).
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No. It should _decrease_ the average DNS response time.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No. It should (slightly) _decrease_ the usage of the network, by
+keeping most DNS traffic local rather than cross-node.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+No change from existing service/proxy behavior.
+
+###### What are other known failure modes?
+
+None known
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+If enabling node-level topology makes any services slower then that is
```

Comment:
This question means, "as an operator what do I do if you are failing SLOs?". So, say we enable this feature and all of a sudden kube-proxy takes 10 minutes to render iptables. My guess is for this feature, there's not much to be done other than turning it off and filing a bug. But they could for example bump up logging in kube-proxy.